### DISCUSSION TITLE
프로젝트 주제 별 참조 자료 링크

### DISCUSSION THREAD

**Discussion by protocol5** [protocol5/2021ESE-Mooyaho_team#5] [Ideas]
## 해당 이슈는 주제 또는 개발 파트 별 참조 자료 공유를 목적으로 하는 저장 공간입니다.

- [x] 다음과 같은 간단한 양식이 존재합니다. **양식을 준수하여 제출하여 주세요.**

> 주제 또는 개발 파트 명(간단하게 서술) : 참조 자료 링크 목적, 자료 링크(하이퍼 링크 권장),참조 파일 번호, @멘션

- 드래그앤 드롭으로 파일을 올릴 수 있습니다. 참조 파일 업로드시 캡션 번호 입력 부탁드립니다.
- 자료에는 동영상, 이미지, 텍스트, 논문 어떤것도 가능합니다.
- 기존에 작성했던 코멘트라면 edit을 통해 추가해주세요.(깔끔한 관리를 위한 권장)
- 코멘트는 삭제하지 말아주세요. 
- 멘션 기능을 통해 관련 이슈자를 선택해주시면 좋습니다.(권장)
- 자세한 예시는 아래 답글을 확인해 주세요.

@DongJune96 @ckdrkt2 @han45699

**Comment by protocol5** [comment]
스마트 미러를 활용한 수화: 손가락 및 손 인식, [유튜브](https://www.youtube.com/watch?v=KtfhmpNvY7Y), 참조사진1, @protocol5 

[참조사진1]
![손 모션 인식](https://user-images.githubusercontent.com/68097267/110585876-a6f36f00-81b4-11eb-8ed4-4c28bf08cfcf.PNG)

**Comment by protocol5** [comment]
이미지 인식 관련 주제 : 인식 프레임 증가하는 방법, resize 함수를 통해 이미지 크기를 줄여 로드 시간을 줄인다.
rgb color scale에서 grayscale로 변환후에 이미지 연산을 하면 속도가 빠르다. (추가시킴 한관우)

**Comment by han45699** [comment]
## 객체 인식 알고리즘 + 라이다를 이용한 전동킥보드 제동장치 및 경고음 장치

- 카메라로 영상 데이터 받아서 객체 인식 알고리즘을 통해 앞에 있는 객체 인식
- 추가적인 이미지 처리를 통해 물체와의 거리 계산
- 거리가 임의로 지정한 threshold보다 가까우면 경고음 출력 및 제동장치 작동
- 이미지 처리 뿐만 아니라 라이다 데이터를 최후의 안전 수단으로 사용

**Comment by DongJune96** [comment]
## 자전거나 킥보드에 장착할 수 있는 스마트 라이트

[참조사진1]
![image](https://user-images.githubusercontent.com/68097144/110588544-80cfce00-81b8-11eb-97f8-f0cfed79afc2.png)
[참조사진2]
![image](https://user-images.githubusercontent.com/68097144/110588609-97762500-81b8-11eb-8d73-17eee5bd9c8c.png)

- 야간에 킥보드나 자전거를 안전하게 타기 위해선 라이트를 멀리까지 비춰 도로의 요철이나 보행자등을 미리 인지하고 속도를 줄이던가 회피를 해야함

- 하지만 보행자와 함께 킥보드나 자전거를 많이 타게되는 특성상 라이트를 멀리까지 비추는데 문제가 있음 (실제로 안양천에서 밤에 자전거 타다 도로가 급격히 꺾이는 구간을 인지하지 못하고 사고날 뻔함)

- 따라서 lidar나 vision을 자전거나 킥보드에 장착해 보행자에 방해가 되지 않도록 라이트를 멀리까지 비출 수 있도록 한다면 도움이 될것으로 보임 (헬멧에 장착해도 될듯)

[출처](https://www.youtube.com/watch?v=75qDwbsT9iY)

**Comment by DongJune96** [comment]
라즈베리파이 프로젝트 참조 사이트
https://www.hackster.io/raspberry-pi/projects
https://nevonprojects.com/raspberry-pi-projects/
https://pimylifeup.com/category/projects/

**Comment by DongJune96** [comment]
## 의식의 흐름

- 라즈베리파이에 무게, 바코드 센서를 장착해 카트에 물건을 넣으면 자동 계산
 ==> 벌써 생각보다 많이 발전된듯 이마트에서 사람 따라다니는 자율주행 카트 시연영상도있음
 아마존에서도 개발중인듯

- 신기해서 가져와봄
![image](https://user-images.githubusercontent.com/68097144/110796610-7e09d180-82bb-11eb-82de-2cbf724518c9.png)
출처 : https://www.hackster.io/kemfic/curved-lane-detection-34f771
관우가 얘기한 제동 및 안전장치랑 연결할 수도 있을거같긴한데 잘 모르겠음

- 동전 분류
 ==> 동전 분류는 너무 쉬울거같긴함 알약을 분류해서 조제?
동전 분류 : https://courses.ece.cornell.edu/ece5990/ECE5725_Spring2020_Projects/May_18_Demo/Coin%20Sorter/yw482_zl385_Wednesday/index.html

- 스마트 홈(방 안의 대기질이나 온도 등을 파악해 에어컨 공기청정기 등 작동)이나 스마트 도어 스마트미러 이런 주제들이 그동안 프로젝트로 많이 진행된 듯

**Comment by han45699** [comment]
## 아마존 Alexa와 라즈베리 파이를 이용한 노크한 사람 인지 + 사람데이터 저장 상호작용

https://www.hackster.io/gmelaub/alexa-who-s-at-the-door-d0f6b5#

- **기존 알고리즘**
1. 누가 노크를 하면, 집 안에 있는 사람이 알렉사를 부른다.
2. 문 앞에 누가 있어? 하고 물어보면 알렉사가 사진을 찍고 이미지 처리를 통해 누구인지 인식한다.
3. 만약 아는 사람이면 그 사람 이름을 알려준다.
4. 만약 모르는 사람이면 사진을 찍을까요? 하고 물어본다. 이후 사용자가 이름 데이터를 말해주면 그 이름으로 새로운 사람을 저장한다.
5. 데이터를 쌓으면 쌓을 수록 인지 정확도가 상승한다.




- **추가하고 싶은 아이디어**
1. 라즈베리 파이가 문에 부착되어있게 만든 다음, 노크가 울리면(가속도계를 사용하던지) 자동적으로 집 안에 있는 사람에게 (아는 사람이면 '홍길동'이 노크했습니다.) 또는 (모르는 사람이면 '모르는 사람'이 노크를 했습니다.) 하고 알려주기
2. 상호작용 하는 과정에서 사용자가 통화하고 싶다고 말하면 모니터에 캠 띄우고 음성 통화 가능하게 하기



----------------------------------------------------------------
비슷한 프로젝트
https://www.hackster.io/SeeedStudio/face-recognization-smart-lock-with-lte-pi-hat-abcec9

**Comment by han45699** [comment]
## 도어벨 누르면 집안의 사용자 스마트폰에 이메일이 가서, 이메일에 접속하면 서로 영상통화가 되게 하는 + 문까지 열 수 있는  프로젝트 
https://www.hackster.io/hackershack/smart-doorbell-video-intercom-system-e5aa61

**Comment by ckdrkt2** [comment]
# 가위 바위 보 로봇
![ezgif com-gif-maker](https://user-images.githubusercontent.com/72551588/110811534-22931000-82ca-11eb-8f53-4bf77af6580d.gif)
https://projects.raspberrypi.org/en/projects/rock-paper-scissors-by-hand

- 카메라로 손의 모양을 인식하여 가위 바위 보 게임을 진행하는 로봇
- 아쉽게도 코드를 제공하는 것 같지는 않고 Techable Machine 이라는 사이트를 통해서 서비스 제공
- 손 또는 손가락 인식 주제(수화 등)에 응용하거나 적용할 수 있는 방법을 찾으면 도움이 될 거라고 생각
- 같은 사이트 내에 있는 이미지 인식 관련 내용 링크
https://projects.raspberrypi.org/en/projects/amazing-image-identifier

**Comment by ckdrkt2** [comment]
# 라즈베리파이4로 토이프로젝트용 서버 만들기
https://ryan-han.com/post/server/raspberry_server_1/

- 라즈베리 파이 4와 추가적으로 필요한 제품(쿨링 팬 등)을 사용하여 간단한 서버 구축
- IoT과 같은 주제를 사용하게 되었을 때 활용할 수 있을 것이라고 생각

**Comment by ckdrkt2** [comment]
# 라즈베리 파이 얼굴 인식 with Python, OpenCV, Deep learning
![제목 없음](https://user-images.githubusercontent.com/72551588/110804589-ab5a7d80-82c3-11eb-93e9-96c472703cf2.jpg)
https://www.pyimagesearch.com/2018/06/25/raspberry-pi-face-recognition/

- 인식 과정
![image](https://user-images.githubusercontent.com/72551588/110810017-bc59bd80-82c8-11eb-918e-ccda04b8609f.png)
- 라즈베리 파이로 얼굴 인식에 대한 가이드 제공
- 전체적인 과정과 코드 구조에 대해 설명이 되어있고 전체 코드도 다운받을 수 있음
- 소스 코드 외에도 17 페이지 정도 되는 가이드를 제공(이메일 작성 시)
- 이외에도 pyimagesearch에 라즈베리 파이 관련되어 사용할 수 있는 OpenCV에 대한 내용들이 있음
[https://www.pyimagesearch.com/?s=raspberry+pi+opencv](url)

**Comment by protocol5** [comment]
## 다중 무선 충전 패드 - 샤오미

- 여러 전자기기를 패드위에 올리면 기기의 좌표를 찾아 충전 단자가 찾아감
- 모든 제품의 충전을 지원
- 구성이 간단하여 구현 가능하나 만약 주제로 된다면 개선사항이 필요
- 단점으로는 모터 소음과 좌표를 찾아가는 시간
![image](https://user-images.githubusercontent.com/68097267/110817337-8c61e880-82cf-11eb-824c-a936c1b23676.png)
[유튜브 링크](https://www.youtube.com/watch?v=nqfbGxIaVgs)

**Comment by DongJune96** [comment]
## 시계

![image](https://user-images.githubusercontent.com/68097144/110819908-faa7aa80-82d1-11eb-96b4-736d26cf3a98.png)
![image](https://user-images.githubusercontent.com/68097144/110819951-02674f00-82d2-11eb-8528-94eef37d0eeb.png)
[출처](https://www.youtube.com/watch?v=1GBGIU9TGgo)
- 여러개의 작은 아날로그 시계들이 디지털 시계 처럼 동작

- 시간을 표시하는 것 뿐만 아니라 특정 문구나 그림 등 사용자가 원하는 것을 출력하도록 한다면 괜찮을듯 

![image](https://user-images.githubusercontent.com/68097144/110820722-b49f1680-82d2-11eb-92d9-5f5ab3b955c0.png)
-이런식으로 특정 시간마다 시간을 펜을 이용해 썼다 지웠다하는 비효율적인? 프로젝트도있음
[출처](https://courses.ece.cornell.edu/ece5990/ECE5725_Fall2019_Projects/Dec_10_Demo/Pi%20Clock/rd568_wf227_Thursday/index.html)

**Comment by protocol5** [comment]
유태영 comment : 라즈베리파이사용시 fps낮게나옴, 젯슨보드를 tcp/ip로 붙여도 최대 30fps 밖에 안나옴.

**Comment by han45699** [comment]
No.1 | 적외선 센서를 이용한 비접촉 엘리베이터 스위치+얼굴인식 | 고동준 | 비고
-- | -- | -- | --
  | [요약] 코로나 시대에 비접촉을 위한 버튼 방식이 아닌 적외선 센서와 안면 인식을 이용한 엘리베이터 스위치 버튼 방식이 아닌 적외선 센서와 안면 인식을 이용한 엘리베이터 스위치 + 사용자의 높이에 따라 이동  [장점]  - 코로나로 인해 비접촉이 대두화되는 시기에 실제 물리적으로 누르는 버튼보다 안전하게 사용 가능  - 안면인식을 통해 기존의 카드키를 대면 해당 층으로 이동하는 것보다 개선가능  - 아이, 어른 등 다양한 사용자의 눈높이와 키높이에 맞춰 자동으로 레일을 이용해 자동으로 이동  [구동 알고리즘]   1. 사람이 엘리베이터에 탑승하면, 안면 인식 진행  2. 등록된 사용자이면 등록된 층으로 자동 이동  3. 등록되지 않은 사용자이면, 층수 입력  4. 아래의 사진과 같이  손가락으로 이동하고자하는 층을 입력  5. 적외선센서 또는 카메라를 이용해 손가락의 움직임으로 층수 판별 |   |

**Comment by protocol5** [comment]
웹캠 설정 https://webnautes.tistory.com/909

**Comment by ckdrkt2** [comment]
# 최종 주제 관련 조사

1. Finger Detection and Tracking using OpenCV and Python
![image](https://user-images.githubusercontent.com/72551588/111884452-106d4c00-8a05-11eb-9b05-e08b11d83b2a.png)
https://dev.to/amarlearning/finger-detection-and-tracking-using-opencv-and-python-586m

- 피부색과 히스토그램을 이용하여 손을 인식하고 마스킹으로 경계를 나눔
- 이미지 형상 분리는 cv2.calcBack Project 를 사용
- 손가락 끝 인식을 위해 마스킹 한 경계에 등고선을 그리고 등고선의 중심에서 가장 멀리 떨이진 볼록한 부분을 확인
- 볼록한 부분들을 결점으로 간주하고 가장 멀리 떨어져있는 결점을 손가락 끝으로 인식

2. Hand-Tracking For Smartphone AR Apps Is On Its Way
![KakaoTalk_20210320_155123539](https://user-images.githubusercontent.com/72551588/111885099-fd5c7b00-8a08-11eb-9d1f-031a855624bd.gif)
https://www.vrfocus.com/2018/06/hand-tracking-for-smartphone-ar-apps-is-on-its-way/

3. Ball Tracking with OpenCV
![image](https://user-images.githubusercontent.com/72551588/111884527-870a4980-8a05-11eb-97eb-ed15000bc5f5.png)
https://www.pyimagesearch.com/2015/09/14/ball-tracking-with-opencv/

4. Finding extreme points in contours with OpenCV
![image](https://user-images.githubusercontent.com/72551588/111884968-36482000-8a08-11eb-835b-bbb6fcfc11d2.png)
https://www.pyimagesearch.com/2016/04/11/finding-extreme-points-in-contours-with-opencv/

5. OpenCV People Counter
![opencv_people_counter_result01](https://user-images.githubusercontent.com/72551588/111885070-dc942580-8a08-11eb-9a20-49c10a72a8a1.gif)
https://www.pyimagesearch.com/2018/08/13/opencv-people-counter/

- 카메라로 사람의 입출입을 카운트하는 OpenCV
- 엘리베이터 또는 건물 보안과 관련되어 사용할 수 있을 것으로 생각되어서 추가함

**Comment by DongJune96** [comment]
# 자동으로 컵이나 냄비의 형상을 인식해 원하는 양의 물만 따라주는 정수기

[선행 연구]
2019년 LG전자에서 발표된 상하좌우정수기
   - https://www.youtube.com/watch?v=8yq-4yGJiYY
   - 물이 나오는 부분이 밑으로 내려오며 컵에 닿으면 조금 올라간 후 물이 나옴
   - 밑으로 계속 내려오다 컵에 닿지 않으면 내려갈 수 있는 가장 아래에서 물이 나옴
   - 따라서 컵의 형상이나 높이를 카메라 같은 것으로 인식하는 것은 아님

[설계 방안]
  - 카메라나 적외선 센서를 이용해 컵의 형상 인식해  사용자가 미리 설정해 둔 높이만큼 물을 채우도록  하는 정수기를 만든다면 좋을듯  (컵의 80%, 50% 등등)

**Comment by DongJune96** [comment]
# 버스 정류장 자동 통과 시스템

[문제점]
- 버스 탑승객중 다음 정거장에 내릴 사람이 없는 상황을 가정. 현재는 다음 정류장에 해당 버스를 탑승할 사람이 없어도 버스기사는 이를 모르기 때문에 어쩔 수 없이 정류장에 가서 버스를 세워야함. 이 과정에서 차선변경과 정차가 이뤄지므로 교통체증의 원인이 됨.
- 특히 야간이나 이른 시간에는 버스 이용자가 많이 없지만 정류장마다 서야하기 때문에 매우 비효율적

[설계 방안]
- 현재 정류장에 있는 버스 도착 정보 모니터를 이용해 승객이 해당 버스를 탑승할지 안할지를 선택해 버스 기사에게 알려 굳이 정차를 하지 않아도 될 상황이라면 그냥 지나가도록 함

- 특정 버스를 선택하는 것이 아니라 목적지(서울시립대 입구 정류장) 등을 선택하게 해 해당 노선을 지나가는 버스 중 맨 앞의 버스만 정차하도록 한 후 나머지 버스는 그냥 통과하도록함
- 음성인식 등을 이용해 목적지를 말하면 자동으로 정류장에 도착할 버스들에 정보를 전송해 컨트롤하게 할 수도 있음

**Comment by protocol5** [comment]
### [주제] 웨어러블 위치센서와 화상 통신을 통한 노약자 상태 확인 시스템- 오정규 

 

1. 개요 및 동기 

* 노약자 가족을 두고있는 경험과 입장으로 항상 마음이 불안하지만, 지속적으로 연락을 취하거나 24시간 모니터링을 하는건 여러가지 문제점을 가지고 있음. 그러한이유로 노약자에게 언제 발생할지 모르는 불안정하거나 위험한 상황에 대한 준비는 큰 고민거리였음. 

* 노약자는 대부분 첨단기기 사용에 무지하거나 어려워함. (예시,티비리모콘 사용을 어려워 하셨던 어르신들, 스마트폰사용이 어려워 2g폰을 아직 고집하시는 어르신, 애완동물). 그렇기때문에 화상통화를 통한 간단한 안부인사 또는 상태 검사를 하는것도 어려움이 있고, 그 원인은 노약자와 노약자 가족 모두 사용법을 숙지해야하는 문제에 있음. 
* 기존 비슷한 상품들로는 펫모니터링기기(애완동물 상태를 사용자만 체크하고 음성으로 애완동물에게 말정도만 전달함), 스마트폰을 사용한 아이위치 추적, 웨어러블 위치 추적 및 알람기, 물체 위치를 찾는 스마트태그(삼성)정도가 존재하나, 이는 모두 모니터링을 하는 사용자만이 단방향적 통신을 하거나, 양방향 통신이어도 상대 사용자는 사용법에 능숙해야하는 단점이 존재함. 개발되어진 제품이나 기능들은 양쪽 사용자 모두 스마트폰을 활용해야하는 앱이 가장 많았으며 아니면 어느정도의 사용법을 익혀야하는 제품들이였음. 

2. 아키텍처
![image](https://user-images.githubusercontent.com/68097267/112347750-7a8a3780-8d0a-11eb-80ff-537fba7fdaf0.png)

---
디테일한 내용은 원노트 참조

**Comment by ckdrkt2** [comment]
# 노인 케어를 위한 웨어러블 시스템
https://kookmin-sw.github.io/capstone-2020-19/
![image](https://user-images.githubusercontent.com/72551588/112763963-91e65f00-9041-11eb-9a11-a9386969c153.png)
https://www.youtube.com/watch?v=AUiAEV_3ueg

**Comment by DongJune96** [comment]
# fall detection

- https://github.com/qiaoguan/Fall-detection
- https://www.youtube.com/watch?v=3R8STCfmUI4
